npm/design-system/src/components/FileExplorer/FileExplorer.tsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react'
+import cs from 'classnames'
+import { FileNode, FolderNode, TreeNode } from './helpers/makeFileHierarchy'
+
+export interface FolderComponentProps {
+  item: FolderNode
+  depth: number
+  isOpen: boolean
+  onClick: () => void
+}
+
+export interface FileComponentProps {
+  item: FileNode
+  depth: number
+  onClick: (file: FileNode) => void
+}
+
+export interface FileExplorerProps extends React.HTMLAttributes<HTMLDivElement> {
+  files: TreeNode[]
+  fileComponent: React.FC<FileComponentProps>
+  folderComponent: React.FC<FolderComponentProps>
+  selectedFile?: string
+  onFileClick: (file: FileNode) => void
+
+  // Styles. They should be a *.module.scss.
+  // TODO: Can we type these? Do we want to couple to CSS modules?
+  cssModule?: {
+    nav: any
+    ul: any
+    li: any
+    a: any
+    isSelected: any
+  }
```

Comment:
I'm unsure if this is from Jess's work, but this is quite an unusual pattern. It's fine if we actually need to customize the styles of every little thing, but why would we need to? Additionally, why are these not contextual names?

Reply:
Yeah just left over from the first round through.